pkg/controller/lifecycle/actuator.go

```diff
@@ -111,6 +113,21 @@ func (a *Actuator) GenerateSecretData(logger logr.Logger, ctx context.Context, e
 	return secretData, nil
 }
 
+func (a *Actuator) GenerateSecretDataToMonitoringManagedResource(namespace string) (map[string][]byte, error) {
+	emptyMap := map[string][]byte{}
+
+	// Replace marker in namespace field to true namespace.
+	yamlStringConfigNameWithNamespace := regexp.MustCompile(`{{ namespace }}`).ReplaceAllString(string(consts.MonitoringYaml), namespace)
+
+	// Convert yaml form to json.
+	jsonConfigMap, err := yaml.YAMLToJSON([]byte(yamlStringConfigNameWithNamespace))
```

Comment:
why this convension is needed? in golang "domain" it is just []byte. Does it matter for some other commponents (GRM)? - and as I understand json is just subset of yaml, so why do we need this conversion?

Reply:
You are right, we don't need this conversion. I removed this.